crates/sp-domains-fraud-proof/src/verification.rs

```diff
@@ -270,3 +272,73 @@ where
 
     Ok(())
 }
+
+pub fn verify_true_invalid_bundle_fraud_proof<CBlock, DomainNumber, DomainHash, Balance>(
+    bad_receipt: ExecutionReceipt<
+        NumberFor<CBlock>,
+        CBlock::Hash,
+        DomainNumber,
+        DomainHash,
+        Balance,
+    >,
+    true_invalid_fraud_proof: &TrueInvalidBundleEntryFraudProof,
+) -> Result<(), VerificationError>
+where
+    CBlock: BlockT,
+{
+    let true_invalid_bundle_entry = bad_receipt
+        .inboxed_bundles
+        .get(true_invalid_fraud_proof.bundle_index as usize)
+        .ok_or(VerificationError::BundleNotFound)?;
+
+    if !true_invalid_fraud_proof.matches_with_bundle_entry(true_invalid_bundle_entry) {
+        return Err(VerificationError::FraudProofMismatch);
+    }
+
+    let extrinsic = StorageProofVerifier::<BlakeTwo256>::get_decoded_value(
```

Comment:
Do not use bare hash functions, you the one from DomainBlock or DomainHeader

Reply:
It is not possible here to be generic as the calculation done in `domain_bundle_proposer.rs` is as follows: 
```
let extrinsics_root = BlakeTwo256::ordered_trie_root(
            extrinsics.iter().map(|xt| xt.encode()).collect(),
            sp_core::storage::StateVersion::V1,
        );
```